velox/dwio/text/tests/writer/TextWriterTest.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/dwio/text/writer/TextWriter.h"
+#include <gtest/gtest.h>
+#include "velox/common/base/Fs.h"
+#include "velox/common/file/FileSystems.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+namespace facebook::velox::text {
+std::vector<std::vector<std::string>> readFile(const std::string& name) {
+  std::ifstream file(name);
+  std::string line;
+  std::vector<std::vector<std::string>> table;
+
+  while (std::getline(file, line)) {
+    std::stringstream ss(line);
+    std::string cell;
+    std::vector<std::string> row;
+
+    while (std::getline(ss, cell, TextWriter::SOH)) {
+      row.push_back(cell);
+    }
+
+    table.push_back(row);
+  }
+  return table;
+}
+
+class TextWriterTest : public testing::Test,
+                       public velox::test::VectorTestBase {
+ public:
+  void SetUp() override {
+    velox::filesystems::registerLocalFileSystem();
+    dwio::common::LocalFileSink::registerFactory();
+    rootPool_ = memory::memoryManager()->addRootPool("TextWriterTests");
+    leafPool_ = rootPool_->addLeafChild("TextWriterTests");
+    tempPath_ = exec::test::TempDirectoryPath::create();
+  }
+
+ protected:
+  static void SetUpTestCase() {
+    memory::MemoryManager::testingSetInstance({});
+  }
+
+  std::shared_ptr<memory::MemoryPool> rootPool_;
+  std::shared_ptr<memory::MemoryPool> leafPool_;
+  std::shared_ptr<exec::test::TempDirectoryPath> tempPath_;
+};
+
+TEST_F(TextWriterTest, write) {
+  auto schema = ROW({"c0", "c1"}, {BIGINT(), BOOLEAN()});
```

Comment:
I think we shall cover all the supported data types here and verify data correctness.